### PR TITLE
add new rp2biosensor v 1.0.1

### PR DESCRIPTION
**Pull request by kenza12**
**rp2biosensor** : Build Sensing-Enabling Metabolic Pathways from RetroPath2.0 output

Development repository : https://github.com/brsynth/rp2biosensor

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

**Comment by tduigou** [comment]
Hi, I confirm I'm a maintainer of the package.
Best wishes,

File: recipes/rp2biosensor/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{% set name = "rp2biosensor" %}
+{% set version = "1.0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/brsynth/rp2biosensor/archive/refs/tags/{{ version }}.tar.gz
```
Comment (bollwyvl) [inline recipes/rp2biosensor/meta.yaml:9]: i'm not an official reviewer, but: if at all possible, following the general trend of a "canonical" upstream source release on, in this case, PyPI, would be preferable to use of a github source tarball. This helps keep automated tooling that relies on this namespace up-to-date.

File: recipes/rp2biosensor/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  source_files:
+    - tests
+  imports:
+    - rp2biosensor
+  requires:
+    - pytest
+  commands:
+    - python -m rp2biosensor --help
```
Comment (bollwyvl) [inline recipes/rp2biosensor/meta.yaml:35]: you'll want to add `pip` to requires and run `pip check` as well. This helps ensure that your upstreams are _also_ up-to-date with their metadata.

Comment (bollwyvl) [reply recipes/rp2biosensor/meta.yaml:35]: also: if you're going to make an actual command line tool, in `setup.py` go ahead and make an `entry_points={"console_scripts": ["rp2biosensor = rp2biosensor:main"]}` or something along those lines. That needs to get hoisted in the `build` section of _this_ file: https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#python-entry-points

This helps ensure that all of the plumbing is set up properly to make cross-platform scripts, correctly added to the path, etc.

File: recipes/rp2biosensor/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipes/rp2biosensor/meta.yaml:9]: As part of that, you'd want to ensure the license file gets included in the distribution, by adding `include LICENSE.md` to `MANIFEST.in`

File: recipes/rp2biosensor/meta.yaml (hunk 2 above)
Comment (kenza12) [reply recipes/rp2biosensor/meta.yaml:39]: Hi @bollwyvl ,

Thanks for your suggestions.

Sorry but we don't have PyPI packages, only sources from the github.

We could make some changes:
- add entry_points to the recipe and setup.py
- add pip check to the recipe
- include license.md to MANIFEST.in

Thank your for you work.

Best wishes.

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

File: recipes/rp2biosensor/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  script: {{ PYTHON }} -m pip install -vv .
+  entry_points:
+    - rp2biosensor = rp2biosensor.__main__:main
+
+requirements:
+  host:
+    - python >=3.7,<3.9
+    - pip
+  run:
+    - python >=3.7,<3.9
```
Comment (jaimergp) [inline recipes/rp2biosensor/meta.yaml:24]: Are these upper bounds necessary?

Comment (kenza12) [reply recipes/rp2biosensor/meta.yaml]: Hi @jaimergp ,

Indeed, it's not necessary, we changed that to `python >=3.6`. Thank you.

Comment (jaimergp) [reply recipes/rp2biosensor/meta.yaml]: `setup.py` [says](https://github.com/brsynth/rp2biosensor/commit/44fc21c1f315a35a30067108f28751ff9d36655b#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R45) you need Python 3.7+. Which one is correct?

Comment (kenza12) [reply recipes/rp2biosensor/meta.yaml]: @jaimergp , Python3.6+ is the correct one. We just modified the setup too. Thanks again.

**Comment by jaimergp** [comment]
Excellent, thanks!

**Pull request merged**